src/Essentials/test/DeviceTests/Tests/Permissions_Tests.cs

```diff
@@ -103,5 +103,38 @@ public async Task StorageAndroid13AlwaysGranted()
 				Assert.Equal(PermissionStatus.Denied, status);
 			}
 		}
```

Comment:
To ensure the fix works and avoids regressions, add unit tests that simulate the coarse-only state and verify that `CheckStatusAsync` returns `Restricted`.